crates/db_schema/src/source/community.rs

```diff
@@ -189,52 +188,53 @@ pub struct CommunityUpdateForm {
 #[cfg_attr(feature = "full", diesel(table_name = community_actions))]
 #[cfg_attr(feature = "full", diesel(primary_key(person_id, community_id)))]
 #[cfg_attr(feature = "full", diesel(check_for_backend(diesel::pg::Pg)))]
-pub struct CommunityModerator {
+#[cfg_attr(feature = "full", ts(export))]
+pub struct CommunityActions {
   pub community_id: CommunityId,
   pub person_id: PersonId,
-  #[cfg_attr(feature = "full", diesel(select_expression = community_actions::became_moderator.assume_not_null()))]
-  #[cfg_attr(feature = "full", diesel(select_expression_type = dsl::AssumeNotNull<community_actions::became_moderator>))]
-  pub published: DateTime<Utc>,
+  #[cfg_attr(feature = "full", ts(optional))]
+  /// When the community was followed.
+  pub followed: Option<DateTime<Utc>>,
+  #[cfg_attr(feature = "full", ts(optional))]
+  /// The state of the community follow.
+  pub follow_state: Option<CommunityFollowerState>,
+  #[cfg_attr(feature = "full", ts(optional))]
+  /// The approver of the community follow.
+  pub follow_approver_id: Option<PersonId>,
+  #[cfg_attr(feature = "full", ts(optional))]
+  /// When the community was blocked.
+  pub blocked: Option<DateTime<Utc>>,
+  #[cfg_attr(feature = "full", ts(optional))]
+  /// When this user became a moderator.
+  pub became_moderator: Option<DateTime<Utc>>,
+  #[cfg_attr(feature = "full", ts(optional))]
+  /// When this user received a ban.
+  pub received_ban: Option<DateTime<Utc>>,
+  #[cfg_attr(feature = "full", ts(optional))]
+  /// When their ban expires.
+  pub ban_expires: Option<DateTime<Utc>>,
```

Comment:
A lot of the actions are stored as timestamp when they could be simple bool, not sure whats the reason for that. Even if it should be timestamp in the db for some reason, it could be converted to bool for the api.

Reply:
@dullbananas would have to chime in there, I think timestamps are probably preferable to bools anyway. Clients might want to display *when* the action was done, instead of just that it was done.